server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -1314,6 +1320,11 @@ private void parseSource(DefaultSearchContext context, SearchSourceBuilder sourc
             context.evaluateRequestShouldUseConcurrentSearch();
             return;
         }
+        // Can be marked false for majority cases for which star-tree cannot be used
+        // Will save checking the criteria later and we can have a limit on what search requests are supported
+        // As we increment the cases where star-tree can be used, this can be set back to true
+        boolean canUseStarTree = context.mapperService().isCompositeIndexPresent();
```

Comment:
Should this be based on dimensions and metric rather ?

Reply:
This is just initialization, will toggle this off for multiple cases. (TODO-in this PR iteself)